### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by wsnoble** [crux-toolkit/crux-toolkit#579]
This tide-search command worked fine, but then I switched from --exact-p-value T to --use-tailor-calibration T and I got a bad_alloc error

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

```
crux tide-search --use-tailor-calibration T --isotope-error 1,2 --mz-bin-width 1.0005079 --auto-precursor-window warn --num-threads 8 --fileroot narrow --output-dir rep1/C1 ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_101.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_102.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_103.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_104.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_1\
05.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_106.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_107.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_108.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_109.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_110.mzML ../../../data/rep1/Pf_C1-593_MixES-Sol_SG-1_rKCTi_VO2_111.mzML plasmo-human
```

I am using Crux version 4.1-f49179dc-2022-06-29.

[narrow.tide-search.log.txt](https://github.com/crux-toolkit/crux-toolkit/files/9128655/narrow.tide-search.log.txt)

**Comment by wsnoble** [comment]
I re-ran this without tailor calibration, and it also ran out of memory.

**Comment by wsnoble** [comment]
I ran it on a machine with 128G instead of 32G, and it still ran out of memory. 

I also confirmed that the same command runs successfully on a 32G machine using Crux version 3.2-46bb0c1.

**Comment by kfattila** [comment]
fixed with pull request #581
